pvlib/snow.py

```diff
@@ -185,3 +185,134 @@ def dc_loss_nrel(snow_coverage, num_strings):
        Available at https://www.nrel.gov/docs/fy18osti/67399.pdf
     '''
     return np.ceil(snow_coverage * num_strings) / num_strings
+
+
+def _townsend_effective_snow(snow_total, snow_events):
+    '''
+    Calculates effective snow using the total snowfall received each month and
+    the number of snowfall events each month.
+
+    Parameters
+    ----------
+    snow_total : array-like
+        Snow received each month. Referred to as S in [1]_ [cm]
+
+    snow_events : array-like
+        Number of snowfall events each month. Referred to as N in [1]_ [-]
+
+    Returns
+    -------
+    effective_snowfall : array-like
+        Effective snowfall as defined in the Townsend model [cm]
+
+    References
+    ----------
+    .. [1] Townsend, Tim & Powers, Loren. (2011). Photovoltaics and snow: An
+       update from two winters of measurements in the SIERRA. Conference
+       Record of the IEEE Photovoltaic Specialists Conference.
+       003231-003236. :doi:`10.1109/PVSC.2011.6186627`
+    '''
+    snow_events_no_zeros = np.maximum(snow_events, 1)
+    effective_snow = 0.5 * snow_total * (1 + 1 / snow_events_no_zeros)
+    return np.where(snow_events > 0, effective_snow, 0)
+
+
+def loss_townsend(snow_total, snow_events, surface_tilt, relative_humidity,
+                  temp_air, poa_global, slant_height, lower_edge_drop_height,
+                  angle_of_repose=40):
+    '''
+    Calculates monthly snow loss based on the Townsend monthly snow loss
+    model [1]_.
+
+    Parameters
+    ----------
+    snow_total : array-like
+        Snow received each month. Referred to as S in [1]_ [cm]
+
+    snow_events : array-like
+        Number of snowfall events each month. Referred to as N in [1]_ [-]
+
+    surface_tilt : float
+        Tilt angle of the array [deg]
+
+    relative_humidity : array-like
+        Monthly average relative humidity [%]
+
+    temp_air : array-like
+        Monthly average ambient temperature [C]
+
+    poa_global : array-like
+        Monthly plane of array insolation [Wh/m2]
+
+    slant_height : float
+        Row length in the slanted plane of array dimension [m]
+
+    lower_edge_drop_height : float
+        Drop height from array edge to ground [m]
```

Comment:
What about `lower_edge_height` instead? Defined as "Distance from array lower edge to the ground. [m]"